Makefile

```diff
@@ -1,51 +1,77 @@
-OS_NAME := $(shell uname -s | tr A-Z a-z)
-EXEC = bin/LRez
-ifeq ($(OS_NAME),linux)
-        SHLIB_EXT=so
-else
-        SHLIB_EXT=dylib
+ifeq ($(SHLIB_EXT),)
```

Comment:
In case `SHLIB_EXT` is already set, one does not need to run `uname`. This helps, e.g., with cross-compilation.